waiter/src/waiter/handler.clj

```diff
@@ -232,22 +232,31 @@
   "Retrieves the list of services viewable by the currently logged in user.
    A service is viewable by the run-as-user or a waiter super-user."
   [entitlement-manager query-state-fn prepend-waiter-url service-id->service-description-fn service-id->metrics-fn request]
-  (let [current-state (query-state-fn)]
-    (let [request-params (-> request ru/query-params-request :query-params)
+  (let [{:keys [all-available-service-ids service-id->healthy-instances service-id->unhealthy-instances]} (query-state-fn)]
+    (let [{:strs [run-as-user token token-version] :as request-params} (-> request ru/query-params-request :query-params)
           auth-user (get request :authorization/user)
-          run-as-user-param (get request-params "run-as-user")
+          run-as-user-predicate (fn [{:keys [service-description service-id]}]
+                                  (if run-as-user
+                                    (some-> (get service-description "run-as-user")
+                                            (= run-as-user))
+                                    (authz/manage-service? entitlement-manager auth-user service-id service-description)))
+          source-token-predicate-factory (fn [entry-key target-value]
+                                           (if target-value
+                                             (fn [{{:strs [source-tokens]} :service-description}]
+                                               (->> source-tokens (map #(get % entry-key)) (some #(= target-value %))))
```

Comment:
It seems useful to be able to filter with wildcards on the token. For example, `*dpo*` to give me all tokens containing `dpo`.

Reply:
Cook allows this kind of filtering on job name

Reply:
Done.